File: Niveau 3/Niveau 3.8/entre_deux.hs
--------------------------------------------------------------------------------
-- Name:        Entre deux
-- Purpose:     France-IOI niveau 3.8
--
-- Author:      Sébastien
--
-- Created:     04/10/2022
--------------------------------------------------------------------------------

import Control.Monad
import Data.List

main :: IO()
main = do  
    n:m:_ <- fmap (map read . words) getLine :: IO [Int]
    ecrireEntiers n m
    
ecrireEntiers :: Int -> Int -> IO()
ecrireEntiers n m
    | n <= m = do
               putStr $ show n ++ " "
               ecrireEntiers (n+1) m
    | otherwise = return ()

File: Niveau 3/Niveau 3.8/entre_deux.py
#-------------------------------------------------------------------------------
# Name:        Entre deux
# Purpose:     France-IOI niveau 3.8
#
# Author:      Sébastien
#
# Created:     04/10/2022
#-------------------------------------------------------------------------------

def ecrire_entiers(n, m):
    if n <= m:
        print(n, end=' ')
        ecrire_entiers(n+1, m)

if __name__ == '__main__':
    n, m = map(int, input().split())
    ecrire_entiers(n, m)
